lib/puppet/application/device.rb

```diff
@@ -315,10 +315,10 @@ def main
               Puppet.info _("retrieving resource: %{resource} from %{target} at %{scheme}%{url_host}%{port}%{url_path}") % { resource: type, target: device.name, scheme: scheme, url_host: device_url.host, port: port, url_path: device_url.path }
               resources = find_resources(type, name)
               if options[:to_yaml]
-                text = resources.map do |resource|
-                  resource.prune_parameters(:parameters_to_include => @extra_params).to_hierayaml.force_encoding(Encoding.default_external)
-                end.join("\n")
-                text.prepend("#{type.downcase}:\n")
+                data = resources.map do |resource|
+                  resource.prune_parameters(:parameters_to_include => @extra_params).to_hiera_yaml_hash
+                end.inject(:+)
```

Comment:
Here `data` is a Hash, and if there is more than one key (as when running `puppet resource user`), then the inject will fail with: `Error: Could not run: undefined method `+' for #<Hash:0x000055f111d32428>`. Can we just drop the inject and call `YAML.dump(type.downcase => data)`?

Reply:
Oh, yes, we have an array of hashes which should be merged together…

```yaml
---
user:
  foo: ...
  bar: ...
```

Just passing the array will not to the right thing:

```yaml
---
user:
- foo: ...
- bar: ...
```

I'm on it!